src/main/java/io/strimzi/kafka/bridge/Application.java

```diff
@@ -47,32 +54,75 @@ public static void main(String[] args) {
 
             Map<String, Object> config = ar.result().getMap();
 
+            List<Future> futures = new ArrayList<>();
+
+            Future<Void> amqpFuture = Future.future();
+            futures.add(amqpFuture);
             AmqpBridgeConfig amqpBridgeConfig = AmqpBridgeConfig.fromMap(config);
             if (amqpBridgeConfig.getEndpointConfig().isEnabled()) {
                 AmqpBridge amqpBridge = new AmqpBridge(amqpBridgeConfig);
+
                 vertx.deployVerticle(amqpBridge, done -> {
                     if (done.succeeded()) {
-                        log.debug("AMQP verticle instance deployed [{}]", done.result());
+                        log.info("AMQP verticle instance deployed [{}]", done.result());
+                        amqpFuture.complete();
                     } else {
-                        log.debug("Failed to deploy AMQP verticle instance", done.cause());
+                        log.error("Failed to deploy AMQP verticle instance", done.cause());
+                        amqpFuture.fail(done.cause());
                     }
                 });
+            } else {
+                amqpFuture.complete();
             }
 
+            Future<Void> httpFuture = Future.future();
+            futures.add(httpFuture);
             HttpBridgeConfig httpBridgeConfig = HttpBridgeConfig.fromMap(config);
             if (httpBridgeConfig.getEndpointConfig().isEnabled()) {
                 HttpBridge httpBridge = new HttpBridge(httpBridgeConfig);
                 vertx.deployVerticle(httpBridge, done -> {
                     if (done.succeeded()) {
-                        log.debug("HTTP verticle instance deployed [{}]", done.result());
+                        log.info("HTTP verticle instance deployed [{}]", done.result());
+                        httpFuture.complete();
                     } else {
-                        log.debug("Failed to deploy HTTP verticle instance", done.cause());
+                        log.error("Failed to deploy HTTP verticle instance", done.cause());
+                        httpFuture.fail(done.cause());
                     }
                 });
+            } else {
+                httpFuture.complete();
             }
+
+            CompositeFuture.join(futures).setHandler(done -> {
+                if (done.succeeded()) {
+                    startHealthServer(vertx);
+                }
+            });
         });
     }
 
+    /**
+     * Start an HTTP health server
+     */
+    private static void startHealthServer(Vertx vertx) {
+
+        vertx.createHttpServer()
+                .requestHandler(request -> {
+                    if (request.path().equals("/healthy")) {
+                        request.response().setStatusCode(HttpResponseStatus.OK.code()).end();
+                    } else if (request.path().equals("/ready")) {
+                        request.response().setStatusCode(HttpResponseStatus.OK.code()).end();
```

Comment:
Should this be bound to the Kafka connectivity? Maybe at least as a issue for the future? Or is that a problem from the technical perspective?

Reply:
Yeah I didn't add a fine control just for now but I was going to open an issue to improve the check. We are on the same page.